Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -297,6 +306,9 @@ protected override bool OnBackButtonPressed()
 
 		void InsertPageBefore(Page page, Page before)
 		{
+			if (page == null || before == null)
+				throw new ArgumentException($"Both {nameof(page)} and {nameof(before)} must be initialized.");
```

Comment:
Sorry, not sure how I missed that part. I think I was trying to be consistent with the existing throws. You mean it's not required to check for null on non-public APIs? Or it's not required to validate parameters at all? If it's the latter, `InsertPageBefore` and `RemovePage` were already checking validity.